src/config.ts

```diff
@@ -589,27 +590,45 @@ export const defaultHttpServiceWithRateLimiterConfig: HttpServiceConfig = {
     metaTxnRateLimiters: META_TXN_RATE_LIMITER_CONFIG,
 };
 
+export const INTEGRATOR_KEYED_BY_ID = transformIntegratorsAcl(INTEGRATORS_ACL, 'integratorId');
+
 /**
  * Gets the integrator ID for a given API key. If the API key is not in the configuration, returns `undefined`.
- * `integratorsMap` is closed in so it only needs to be evaluated once. Much efficiency!
  */
 export const getIntegratorIdForApiKey = (
```

Comment:
Ok I looked through this thing for a longgg time trying to make sense of it. I think the things that added the complexity for me were 1) the `byKey` addition to `transformIntegratorsAcl` and 2) the way ApiKey -> Integrator is exported as a function but IntegratorId -> Integrator is a map constant.

I think the `byKey` thing is worth keeping, but for the maps we should stick with 1 pattern. So my humble suggestion is to either:

1) (my preference) export a `getIntegratorForIntegratorId` function similar to `getIntegratorForApiKey`.
2) export a `INTEGRATOR_KEYED_BY_API_KEY` constant `Map`. 

#1 is my slight preference because for the caller of the function gets a nice signature and knows they just need to put in a string and get out `string | undefined`; the implementation using `Map` is hidden. With the constant the caller must know the constant is a `Map` and act accordingly. Ultimately either one works for me.

Reply:
Either way we should probably add a comment if the function/value is being exported.

Reply:
Thanks for your feedback! I'm going to bias towards your suggestion! changes coming up right now!